cluster.go

```diff
@@ -643,9 +646,18 @@ func (c *clusterClient) pickMulti(ctx context.Context, multi []Completed) (*conn
 	return conns, nil
 }
 
+func isMulti(cmd Completed) bool {
+	return len(cmd.Commands()) == 1 && cmd.Commands()[0] == "MULTI"
+}
+func isExec(cmd Completed) bool {
+	return len(cmd.Commands()) == 1 && cmd.Commands()[0] == "EXEC"
+}
+
 func (c *clusterClient) doresultfn(
-	ctx context.Context, results *redisresults, retries *connretry, mu *sync.Mutex, cc conn, cIndexes []int, commands []Completed, resps []RedisResult, attempts int,
+	ctx context.Context, results *redisresults, retries *connretry, mu *sync.Mutex, cc conn, cIndexes []int, commands []Completed, resps []RedisResult, attempts int, single bool,
```

Comment:
`single` is a hint that the original DoMulti is only involved with 1 connection. We use this hint to skip unnecessary tx checks.